tests/scripts/connect-error-using-cluster-nodes-test.sh

```diff
@@ -22,7 +22,7 @@ syncpid=$!
 timeout 5s ./simulated-redis.pl -p 7400 -d --sigcont $syncpid <<'EOF' &
 EXPECT CONNECT
 EXPECT ["CLUSTER", "NODES"]
-SEND "653876bce37106406581ddc05d8629357a223a7e :30001@40001 myself,master - 0 0 0 connected\n"
+SEND "653876bce37106406581ddc05d8629357a223a7e :30001\@40001 myself,master - 0 0 0 connected\n"
```

Comment:
Is this a shell issue? I thought when `<<'EOF'` is used no chars need t be escaped. It depends on the single quotes around EOF.

Or is it a Perl issue?

Reply:
If this string is used as a perl string literal, we can use single quotes around the perl string, i.e. `SEND '6538...'`.

Reply:
I believe its because of the `eval` in simulated-redis.pl
```
    if (/^SEND (.*)/) {
        my $data = $1;
        if ($data =~ /^[-+\$\*:]/) {
            # Redis protocol with character escapes
            # e.g. '-ERR Unknown command: FOO\r\n'
            $data = unescape($data);
            $data .= "\r\n" unless $data =~ /\r\n$/;
        } else {
            # e.g. '["foo", "bar", 42]'
            $data = redis_encode(eval $1);     <-----
        }
```

with `SEND '....\n'` is there a good way to get the needed `\n` as a newline?

Reply:
Eval means the input is Perl code, which is not intuitive.

Idea: We can switch to JSON. `use JSON;` and then replace `eval` with `decode_json`.